server/channels/app/web_broadcast_hooks.go

```diff
@@ -36,6 +38,9 @@ func (h *addMentionsBroadcastHook) Process(msg *platform.HookedWebSocketEvent, w
 	if len(mentions) > 0 && slices.Contains(mentions, webConn.UserId) {
 		// Note that the client expects this field to be stringified
 		msg.Add("mentions", model.ArrayToJSON([]string{webConn.UserId}))
+		if msg.Get("should_ack") != true {
```

Comment:
We are doing a "best guess" here of users that could have a notification triggered. In your example, the check for muted is done on the front-end so since we're not changing any behaviour here yet, we just want to collect everyone who could receive a notification.

This could include anyone who follows, or is mentioned. I didn't want to collect the users twice since that seemed unnecessary, so it was much easier to just add it here.

Reply:
Do we have a guarantee on what order the hooks run? If not, then we might not be ACKing all the right ones if we don't have mentions/followers available.